distributed/shuffle/_core.py

```diff
@@ -306,7 +308,11 @@ def get_output_partition(
         if not self.transferred:
             raise RuntimeError("`get_output_partition` called before barrier task")
         sync(self._loop, self.flush_receive)
-        return self._get_output_partition(partition_id, key, **kwargs)
+        with (
+            context_meter.meter("shuffle-get-output-noncpu"),
```

Comment:
I was a bit hesitant to label this "disk", as this also includes GIL contention.